src/librustc_save_analysis/span_utils.rs

```diff
@@ -409,7 +409,7 @@ impl<'a> SpanUtils<'a> {
 
         // Otherwise, a generated span is deemed invalid if it is not a sub-span of the root
         // callsite. This filters out macro internal variables and most malformed spans.
-        !parent.source_callsite().contains(parent)
+        !parent.source_callsite().contains(sub_span.unwrap())
```

Comment:
Or just change 401-403 to `let sub_span = unwrap_or!(sub_span, return true);` ([`unwrap_or!`](https://github.com/rust-lang/rust/blob/master/src/libsyntax/lib.rs#L61)).